aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -33,16 +35,16 @@ module aptos_framework::block {
     }
 
     /// The `BlockResource` resource is in an invalid state
-    const EBLOCK_METADATA: u64 = 0;
+    const EBLOCK_METADATA: u64 = 1;
     /// An invalid signer was provided. Expected the signer to be the VM or a Validator.
-    const EVM_OR_VALIDATOR: u64 = 1;
+    const EVM_OR_VALIDATOR: u64 = 2;
+    const EINVALID_EPOCH_INTERVAL: u64 = 3;
 
     /// This can only be called during Genesis.
-    public fun initialize_block_metadata(account: &signer, epoch_interval: u64) {
-        timestamp::assert_genesis();
+    public(friend) fun initialize(account: &signer, epoch_interval: u64) {
         system_addresses::assert_aptos_framework(account);
+        assert!(epoch_interval > 0, error::invalid_argument(EINVALID_EPOCH_INTERVAL));
 
-        assert!(!is_initialized(), error::already_exists(EBLOCK_METADATA));
```

Comment:
Unnecessary check as move_to fails if called for the same account and resource more than once. Here and in other modules.

Reply:
Not unnecessary! It is bad practice to let any raw execution failure happen (move_to, move_from, index bounds, arithmetic, etc.). In Diem we had verified with the Move Prover that all failure conditions are clearly visible from the code. Also the error message will be pretty bad.